Tasks/Npm/npmtoolrunner.ts

```diff
@@ -74,4 +86,47 @@ export class NpmToolRunner extends tr.ToolRunner {
         let config = tl.execSync('npm', `config list ${this.dbg ? '-l' : ''}`, options);
         return options;
     }
+
+    private _getDebugLogPath(options?: tr.IExecSyncOptions): string {
+        // check cache
+        options.silent = true;
+        const cacheLocation = tl.execSync('npm', 'config get cache', options).stdout.trim();
+
+        const logs = tl.findMatch(path.join(cacheLocation, '_logs'), '*-debug.log');
+        if (logs && logs.length > 0) {
+            const debugLog = logs[logs.length - 1];
```

Comment:
What if two builds are running and each emitting their own debug log? Seems like a race condition here, we might grab the wrong log.